iceoryx_posh/source/runtime/posh_runtime_impl.cpp

```diff
@@ -419,8 +419,13 @@ PoshRuntimeImpl::findService(const cxx::variant<Wildcard_t, capro::IdString_t> s
     uint32_t numberOfServices = algorithm::min(capacity, numberOfElements);
     for (uint32_t i = 0U; i < numberOfServices; ++i)
     {
-        capro::ServiceDescription service(cxx::Serialization(requestResponse.getElementAtIndex(i)));
-        serviceContainer.push_back(service);
+        auto maybeService =
```

Comment:
depending on the outcome from above, this might also become something like `deserializeResult`